src/microbe_stage/editor/MicrobeEditor.cs

```diff
@@ -1006,6 +1040,17 @@ private void InitEditor()
         gui.SendUndoToTutorial(TutorialState);
     }
 
+    private void HandleIslands()
+    {
+        var islands = GetIslandHexes();
+        foreach (var hex in islands)
+        {
+            var coords = Hex.AxialToCartesian(hex);
+            var inst = placedHexes.First(p => p.Translation == coords);
+            hexesToResetToIslandMaterial.Add(inst);
```

Comment:
I don't think this approach should be used here.
Instead take a look at how this does things:
https://github.com/Revolutionary-Games/Thrive/blob/a2d68b545c31d6173cee64ac0f2836ba31c667b5/src/microbe_stage/editor/MicrobeEditor.cs#L1624
In fact I think that this entire HandleIslands function should be merged in there.
So the logic would be that if a hex is on the islands list, then use the special material, otherwise use the existing logic in UpdateAlreadyPlacedVisuals.

Reply:
This was my original approach, but didn't work.
I don't find any pattern, but some new placed organelles don't get marked as red....

Reply:
I'm pretty confident it should work. I'd much prefer the islands detection to be included in the normal current species rendering, instead of going through the mechanism intended to restore the right material to things the mouse was hovering over.

Reply:
If you move back to that and commit the code (or link me a previous version if you made a public commit when you had that), I'm willing to take a look to see what the problem is.

Reply:
https://github.com/FuJa0815/Thrive/tree/1694_2_disconnected_grids

This is the weird behaviour I meant:
![output](https://user-images.githubusercontent.com/30809803/95769491-40772d00-0cb8-11eb-85c5-171164e413d6.gif)

Reply:
Interesting, I suggest you run a debugger into the callback that is triggered when something is removed and see where it detects the existing hex from. It should be that the callback is called after the organelle is actually removed.

Reply:
Everything seems to be called correctly in UpdateAlreadyPlacedVisuals, the materialOverride gets set to islandMaterial, but the gui doent't update propperly

Reply:
Is it perhaps reacting to the mouse hover hex near it? Or it is being kept in one of the other restore material lists?

Reply:
The materialOverride got overridden in here:
https://github.com/FuJa0815/Thrive/blob/bff705ba42ec0b214db11ae31bb231570128583e/src/microbe_stage/editor/MicrobeEditor.cs#L1219
adding the instance to the hexesToResetToIslandMaterial list in the UpdateAlreadyPlacedVisuals is not the most pretty solution, but one nevertheless

Reply:
I've merged the changes into this PR

Reply:
You could inside UpdateAlreadyPlacedVisuals make sure that the hex isn't in the wrong list of these:
    private List<MeshInstance> hexesToResetToOldMaterial;
    private List<MeshInstance> hexesToResetToValidMaterial;
    private List<MeshInstance> hexesToResetToIslandMaterial;

If it is remove it from others.

Something like:
if(a)
{
currentHex.material = blah;
hexesToReset.Remove(currentHex);
hexesToResetToSomethingElse.Remove(currentHex);
} else if(...)

That isn't very pretty either but it would at least kinda fit in with the intended flow of the logic.